pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,508 @@
+
+			// Update the latency store
+			m.latencyStore.UpdateNodeIPLatencyEntryByKey(destIP, &NodeIPLatencyEntry{
+				SeqID:           entry.SeqID,
+				LastSendTime:    entry.LastSendTime,
+				LastRecvTime:    end,
+				LastMeasuredRTT: rtt,
+			})
+		}
+	}
+}
+
+func (m *NodeLatencyMonitor) pingAll(ipv4Socket, ipv6Socket net.PacketConn) {
+	// Get all node internal/external IP.
+	nodeIPs := m.latencyStore.ListNodeIPs()
+	klog.InfoS("Start to ping all nodes")
+	for name, toIPs := range nodeIPs {
+		for _, toIP := range toIPs {
+			klog.InfoS("Start to ping node", "Node name", name, "Node IP", toIP)
+			if toIP.To4() != nil && ipv4Socket != nil {
+				if err := m.sendPing(ipv4Socket, toIP); err != nil {
+					klog.Warningf("Failed to send ICMP message to node, err: %#v, Node name: %#v, Node IP: %v", err, name, toIP)
+				}
+			} else if toIP.To16() != nil && ipv6Socket != nil {
+				if err := m.sendPing(ipv6Socket, toIP); err != nil {
+					klog.Warningf("Failed to send ICMP message to node, err: %#v, Node name: %#v, Node IP: %v", err, name, toIP)
+				}
+			} else {
+				klog.Warningf("Failed to send ICMP message to node, Node name: %#v, Node IP: %v", name, toIP)
+			}
+		}
+	}
+}
+
+func (m *NodeLatencyMonitor) report() {
+	// Print all connection status for debug
+	// It will be removed when collector is ready
+	klog.InfoS("Finish to ping all nodes")
+	entries := m.latencyStore.ListLatencies()
+
+	if len(entries) == 0 {
+		klog.Info("No latency data to report, skip reporting")
+		return
+	}
+
+	// Convert to NodeIPLatencyStats
+	nodeIPLatencyStats := &cpv1beta.NodeIPLatencyStat{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: m.gatewayConfig.String(),
+		},
+		NodeIPLatencyList: make([]cpv1beta.NodeIPLatencyEntry, 0, len(entries)),
+	}
+
+	for key, entry := range entries {
+		nodeIPLatencyStats.NodeIPLatencyList = append(nodeIPLatencyStats.NodeIPLatencyList, cpv1beta.NodeIPLatencyEntry{
+			GatewayIP:       key,
+			LastMeasuredRTT: entry.LastMeasuredRTT.Microseconds(),
+			LastSendTime:    entry.LastSendTime.UnixMicro(),
+			LastRecvTime:    entry.LastRecvTime.UnixMicro(),
+		})
+	}
+
+	// Report to controller
+	antreaClient, err := m.antreaClientProvider.GetAntreaClient()
+	if err != nil {
+		klog.ErrorS(err, "Failed to get Antrea client")
+		return
+	}
+	_, err = antreaClient.ControlplaneV1beta2().NodeIPLatencyStats().Create(context.TODO(), nodeIPLatencyStats, metav1.CreateOptions{})
+	if err != nil {
+		klog.ErrorS(err, "Failed to report latency data")
+		return
+	}
+}
+
+func (m *NodeLatencyMonitor) Run(stopCh <-chan struct{}) {
+	// Top level goroutine to handle termination
+	go func() {
+		<-stopCh
+	}()
+
+	// Start the monitor loop
+	go m.nodeLatencyMonitorInformer.Informer().Run(stopCh)
+	go m.nodeInformer.Informer().Run(stopCh)
+	go m.monitorLoop(stopCh)
+}
+
+func (m *NodeLatencyMonitor) monitorLoop(stopCh <-chan struct{}) {
+	// Low level goroutine to handle ping loop
+	var currentTicker *time.Ticker
```

Comment:
why call it `currentTicker` instead of just `ticker`? I don't see another ticker object defined in this function?

Reply:
I will update the value of the `ticker` on restart, so I named it `currenTicker`.

Maybe it would be better to just use the `ticker`.